### DISCUSSION TITLE
How to get dirty state of part of nested form, despite removing fields in other parts of form?

### DISCUSSION THREAD

**Discussion by skorphil** [react-hook-form/react-hook-form#11540] [Q&A]
I have form like:
```js
{
  institutions: [
     { // i need to check if there any changes inside City Bank (institutions.0)
        name:"City Bank", 
        assets: [
          {amount:"", currency:""}
          {amount:"", currency:""}
        ]
     },
  {
        name:"Bank of America",
        assets: [
          {amount:"", currency:""}
          {amount:"", currency:""} // removing or adding asset will result in all `institutions.X` being dirty
        ]
     }
  ]
}
```

I need to check if there is any edits in City Bank:
```js
getFieldState('institutions.0', formState);
```
It works fine if i edit any of the nested fields, like `institutions.0.assets.1.amount`

BUT: if i remove or add any field to any of `institutions.X.assets.X`, all `institutions.X` become dirty, despite there is no changes in them
`formState.dirtyFields` will return that everything is dirty, but individual fields are NOT dirty:
```js
...
  {
        name:false,
        assets: [
          {amount:false, currency:false}
          {amount:false, currency:false}
        ]
     }
``` 

How do i check the individual dirty state of `institutions.0`, without being affected by fields number change in others `institutions.X`?

**Comment by skorphil** [comment]
I know i can manually iterate through all nested fields inside of `institutions.X`, but maybe there is other better way of doing this 'natively'?

**Comment by sujinleeme** [comment]
You can iterate two `useFieldArrays` for nested array fields.
Here's the [codesandbox](https://codesandbox.io/p/sandbox/11540-2ccgjj)

In the `FieldArray` component:
```tsx
 const { fields, append, remove, prepend } = useFieldArray({
    control,
    name: "institutions",
  });
```

In the `NestedFieldArray` component: 
```tsx
  const { fields, remove, append } = useFieldArray({
    control,
    name: `institutions.${nestIndex}.assets`,
  });
```
You might need to use `remove` or `append` from the nested `useFieldArray` to ensure only the nested array is modified. Therefore. it won't overwrite the ancestor and children fields.

In the codesandbox, if a user clicks the "Append nested" button, the nested array is appended. In the console, you can check  all `institutions.0.assets.1` fields are `true`.

<img src="https://github.com/react-hook-form/react-hook-form/assets/6011401/d4aa4fb6-22eb-47d3-88af-a7efeb54e64c" width="300px" />
